nautobot/extras/api/views.py

```diff
@@ -290,16 +319,21 @@ def run(self, request, class_path):
 
         job_content_type = ContentType.objects.get(app_label="extras", model="job")
 
-        job_result = JobResult.enqueue_job(
-            run_job,
-            job.class_path,
-            job_content_type,
-            request.user,
-            data=data,
-            request=copy_safe_request(request),
-            commit=commit,
-        )
-        job.result = job_result
+        schedule = input_serializer.data.get("schedule")
+        if schedule:
+            schedule = self._create_schedule(schedule, data, commit, job, job_class, request)
+        else:
+            job_result = JobResult.enqueue_job(
+                run_job,
+                job.class_path,
+                job_content_type,
+                request.user,
+                data=data,
+                request=copy_safe_request(request),
+                commit=commit,
+                schedule=schedule,
```

Comment:
We only get here if `schedule` is None, and that's its default in `enqueue_job` if unset, so not needed here.

```suggestion
```